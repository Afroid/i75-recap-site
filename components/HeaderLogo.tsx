import Link from "next/link";
import Image from "next/image";
import { useLogo } from "@/lib/LogoContext";
import { TestIds } from "@/lib/testIds";

/**
 * HeaderLogo component displays the site logo and dynamic league logo
 * wrapped in a styled Link to the homepage.
 */
export default function HeaderLogo() {
  // Random logo URL from context
  const logoSrc = useLogo();

  return (
    <Link
      href="/"
      className="relative inline-flex items-center justify-center group h-12"
      data-testid={TestIds.HEADER_LOGO_LINK}
    >
      {/* Background Layers */}

      {/* Darker background, slightly wider */}
      <span
        data-testid={TestIds.HEADER_LOGO_BG_DARK}
        className={[
          "absolute top-0 bottom-0 left-[-10px] right-[-20px] bg-green-600",
          "skew-x-[-25deg] -z-20",
          "transition-transform duration-500 ease-in-out",
          "group-hover:scale-x-110 group-hover:bg-green-700"
        ].join(" ")}
      />

      {/* Lighter background, slightly smaller */}
      <span
        data-testid={TestIds.HEADER_LOGO_BG_LIGHT}
        className={[
          "absolute top-0 bottom-0 left-0 right-[-10px] bg-green-500",
          "skew-x-[-25deg] -z-10",
          "transition-transform duration-500 ease-in-out",
          "group-hover:scale-x-105 group-hover:shadow-lg"
        ].join(" ")}
      />

      {/* Foreground flex container holds the static and dynamic logos */}
      <div
        className="lg:pr-0 lg:pl-4 relative z-10 flex items-center gap-2"
        data-testid={TestIds.HEADER_LOGO_CONTAINER}
      >
        {/* Static site name image */}
        <Image
          src="/I75League.png"
          alt="League Name"
          width={88}
          height={11}
          className="drop-shadow-medium"
          data-testid={TestIds.HEADER_LOGO_NAME}
        />
        {/* Dynamic league logo */}
        <Image
          src={logoSrc}
          alt="League Logo"
          width={88}
          height={11}
          className="h-12 w-auto drop-shadow-medium"
          priority
          data-testid={TestIds.HEADER_LOGO_DYNAMIC}
        />
      </div>
    </Link>
  );
}
