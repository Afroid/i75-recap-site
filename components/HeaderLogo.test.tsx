import React from "react";
import { render, screen } from "@testing-library/react";
import HeaderLogo from "@/components/HeaderLogo";

// Stub next/link to render a plain <a> with the passed props
jest.mock("next/link", () => ({
  __esModule: true,
  default: ({
    href,
    children,
    className,
    "data-testid": testId,
  }: {
    href: string;
    children: React.ReactNode;
    className?: string;
    "data-testid"?: string;
  }) => (
    <a href={href} className={className} data-testid={testId}>
      {children}
    </a>
  ),
}));

// Stub next/image to render a plain <img>
// Here I explicitly destructure out the props we know HeaderLogo will pass, plus the data-testid.
jest.mock("next/image", () => ({
  __esModule: true,
  default: ({
    src,
    alt,
    width,
    height,
    className,
    "data-testid": testId,
  }: {
    src: string;
    alt: string;
    width?: number;
    height?: number;
    className?: string;
    "data-testid"?: string;
  }) => (
    <img
      src={src}
      alt={alt}
      width={width}
      height={height}
      className={className}
      data-testid={testId}
    />
  ),
}));

// Stub the useLogo hook to return a URL
jest.mock("@/lib/LogoContext", () => ({
  useLogo: () => "/dynamic-logo.png",
}));

describe("HeaderLogo", () => {
  it("renders the link wrapper, static & dynamic images correctly", () => {
    render(<HeaderLogo />);

    // The <Link> â†’ <a>
    const link = screen.getByTestId("header-logo-link");
    expect(link).toHaveAttribute("href", "/");

    // Static site name image
    const nameImg = screen.getByTestId("header-logo-name");
    expect(nameImg).toHaveAttribute("src", "/I75League.png");
    expect(nameImg).toHaveAttribute("alt", "League Name");

    // Dynamic league logo image from useLogo
    const dynImg = screen.getByTestId("header-logo-dynamic");
    expect(dynImg).toHaveAttribute("src", "/dynamic-logo.png");
    expect(dynImg).toHaveAttribute("alt", "League Logo");
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<HeaderLogo />);
    expect(asFragment()).toMatchSnapshot();
  });
});
