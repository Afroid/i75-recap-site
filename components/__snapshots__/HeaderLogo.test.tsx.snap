// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLogo matches the snapshot 1`] = `
<DocumentFragment>
  <a
    class="relative inline-flex items-center justify-center group h-12"
    data-testid="header-logo-link"
    href="/"
  >
    <span
      class="absolute top-0 bottom-0 left-[-10px] right-[-20px] bg-green-600 skew-x-[-25deg] -z-20 transition-transform duration-500 ease-in-out group-hover:scale-x-110 group-hover:bg-green-700"
      data-testid="header-logo-bg-dark"
    />
    <span
      class="absolute top-0 bottom-0 left-0 right-[-10px] bg-green-500 skew-x-[-25deg] -z-10 transition-transform duration-500 ease-in-out group-hover:scale-x-105 group-hover:shadow-lg"
      data-testid="header-logo-bg-light"
    />
    <div
      class="lg:pr-0 lg:pl-4 relative z-10 flex items-center gap-2"
      data-testid="header-logo-container"
    >
      <img
        alt="League Name"
        class="drop-shadow-medium"
        data-testid="header-logo-name"
        height="11"
        src="/I75League.png"
        width="88"
      />
      <img
        alt="League Logo"
        class="h-12 w-auto drop-shadow-medium"
        data-testid="header-logo-dynamic"
        height="11"
        src="/dynamic-logo.png"
        width="88"
      />
    </div>
  </a>
</DocumentFragment>
`;
